lib/cylc/batch_sys_manager.py

```diff
@@ -695,6 +700,10 @@ def _job_submit_impl(
             # batch_sys.submit should handle OSError, if relevant.
             ret_code, out, err = batch_sys.submit(job_file_path)
         else:
+            env = None
+            if hasattr(batch_sys, "SUBMIT_CMD_ENV"):
+                env = dict(os.environ)
```

Comment:
Why do we need to pass the whole environment in, not just SUBMIT_CMD_ENV - is it just to make the test below work??